src/machines/modelingMachine.ts

```diff
@@ -574,7 +532,6 @@ export const modelingMachine = createMachine(
       },
 
       'Set selection': {
-        target: '#Modeling',
```

Comment:
Note to self, this is an important details, with the target, even a target which is the top level parent state, it would always transition to the parent's initial state, the target needed to be removed for it not to change the sub state-machine's state.